src/core/dbus-execute.c

```diff
@@ -1678,6 +1681,41 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (streq(name, "RuntimeDirectoryPreserve")) {
```

Comment:
hmm, could you also add the counterparts for this to src/shared/bus-unit-util.c's bus_append_unit_property_assignment()? We should really make sure that all new options come with that right-away (we are still missing some prop hookups for older props, and we should fix that too, but let's get this right right-away for the new ones).

Adding this means the new functionality can be tested easily using:

```sh
systemd-run -t -p RuntimeDirectory=foo -p RuntimeDirectoryPreserve=always /bin/sh
```